java/code/src/com/suse/manager/model/attestation/AttestationFactory.java

```diff
@@ -179,4 +181,25 @@ protected Logger getLogger() {
         return LOG;
     }
 
+    /**
+     * Return a list of reports for a given server with filters
+     * @param serverIn the server
+     * @param earliestIn earliest report
+     * @param offsetIn number of reports to skip
+     * @param limitIn maximal number of reports
+     * @return returns a list or reports
+     */
+    public List<ServerCoCoAttestationReport> listCoCoAttestationReports(Server serverIn, Date earliestIn,
+                                                                        int offsetIn, int limitIn) {
+        return getSession()
+                .createQuery("FROM ServerCoCoAttestationReport " +
+                        "WHERE server = :server " +
+                        "AND modified >= :earliest " +
+                        "ORDER BY modified", ServerCoCoAttestationReport.class)
```

Comment:
Wouldn't it be better to use the `created` date as reference instead of the `modified`? The `modified` will change when checks are attested making the ordering of this list not very consistent and predictable.

Moreover,  I think we should sort descending to offer the newer results first.

Reply:
Yes, makes sense. I checked also System Events listing. They also sort decreasing. I will change this.